embassy-net/src/lib.rs

```diff
@@ -46,15 +48,22 @@ use crate::device::DriverAdapter;
 
 const LOCAL_PORT_MIN: u16 = 1025;
 const LOCAL_PORT_MAX: u16 = 65535;
+const MAX_QUERIES: usize = 2;
 
 pub struct StackResources<const SOCK: usize> {
     sockets: [SocketStorage<'static>; SOCK],
+    #[cfg(feature = "dns")]
+    queries: Option<[Option<dns::DnsQuery>; MAX_QUERIES]>,
```

Comment:
I don't think this is ever `None`?

Reply:
It needs to be moved into the socket, so its taken when stack is crated.

Reply:
The socket should borrow it. If you move it, the `Into<ManagedSlice<..>>`impl gives you a `ManagedSlice::Owned` which doesn't work without alloc.

Reply:
First time i used managed slice, so it was a bit confusing. But now it should be good I think.